### PR TITLE
Enhance id to pk mapping

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 

See #9

### CODE REVIEW

File: lib/mssql.js [hunk 1]
```diff
@@ -499,8 +499,8 @@ MsSQL.prototype.toDatabase = function (prop, val, wrap) {
         && typeof val.length === 'number') { //if value is array
         //check if it is an array of string, because in that cause we need to
         // wrap them in single quotes
-        if (typeof val[0] === 'string') {
```
Comment (ritch) [inline lib/mssql.js]: Should we do some sort of type checking? Removing this seems a bit dangerous at first glance.

File: lib/mssql.js [hunk 2]
```diff
@@ -1133,19 +1133,34 @@ MsSQL.prototype.alterTable = function (model, actualFields, actualIndexes, done,
...
+  var props = Object.keys(objModel.properties);
+  for (var i = 0, n = props.length; i < n; i++) {
+    var prop = props[i];
     if (prop === modelPKID) {
-      return;
+      var idProp = objModel.properties[modelPKID];
+      if (idProp.type === Number) {
+        sql.push("[" + modelPKID + "] [int] IDENTITY(1,1) NOT NULL");
+        continue;
+      } else if (idProp.type === String) {
```
Comment (ritch) [inline lib/mssql.js]: Do we normalize `type` into constructors? Just double checking...

File: lib/mssql.js (hunk 1 above)
Comment (raymondfeng) [inline lib/mssql.js]: escape() can handle different types

Comment (ritch) [inline lib/mssql.js]: Sounds good

**Comment by ritch** [comment]
LGTM

File: lib/mssql.js (hunk 2 above)
Comment (raymondfeng) [inline lib/mssql.js]: Yes, we do

**Pull request merged**